src/lib/server/subscriptionUtils.ts

```diff
@@ -27,9 +27,24 @@ export const getStripeCustomerWithSubscriptions = async (uid: string) => {
         expand: ['subscriptions']
     });
 }
+
+export const getAllCustomerSubscriptions = async (uid: string) => {
+    const stripeCustomerId: string | null = (await db.ref(`/users/${uid}/private/stripeCustomerId`).once('value')).val();
+    if (stripeCustomerId === null)
+        return stripeCustomerId;
+    return await stripe.subscriptions.list({
+        customer: stripeCustomerId,
+        expand: ['data.items'],
+        status: 'all'
+    });
+}
+
```

Comment:
If we parse out the subscriptions from the API response here, then we should be dealing with the same `Stripe.Subscription` object that we get in the customer. And then the `hasCustomerSubscribedBefore` function can just take an array of subscriptions as an argument, which is less tightly coupled with the API response:

```suggestion
export const getAllCustomerSubscriptions = async (uid: string) => {
    const stripeCustomerId: string | null = (await db.ref(`/users/${uid}/private/stripeCustomerId`).once('value')).val();
    if (stripeCustomerId === null)
        return stripeCustomerId;
    const subscriptions = await stripe.subscriptions.list({
        customer: stripeCustomerId,
        expand: ['data.items'],
        status: 'all'
    });
    return subscriptions.data
}
```

Reply:
done!